### PR TITLE
Add load_balancing_service to rosdistro

### PR DESCRIPTION
**Pull request by Barry-Xu-2018**
# Please Add This Package to be indexed in the rosdistro.

## Package name:

load_balancing_service

This package helps ROS2 services implement load balancing functionality.  
https://raw.githack.com/Barry-Xu-2018/ros2_load_balancing_service/main/doc/overview.html

## The source is here:

https://github.com/barry-Xu-2018/ros2_load_balancing_service/

# Checks
 - [x] All packages have a declared license in the package.xml
 - [x] This repository has a LICENSE file
 - [x] This package is expected to build on the submitted rosdistro

### CODE REVIEW

File: rolling/distribution.yaml [hunk 1]
```diff
@@ -3208,6 +3208,12 @@ repositories:
       url: https://github.com/adityapande-1995/linux_isolate_process.git
       version: main
     status: maintained
+  load_balancing_service:
```
Comment (fujitatomoya) [inline rolling/distribution.yaml:3211]: just a checking. currently this package can support only rolling? @Barry-Xu-2018

Comment (Barry-Xu-2018) [reply rolling/distribution.yaml:3211]: Yes. This package depends on `class GenericService`  which is only implemented in rolling.

**Review by Yadunund — COMMENTED** [review]
### New package review checklist

- [x] At least one of the following must be present
  - [x] Top level license file:
  - [ ] Per package license files:
- [x] License is [OSI-approved](https://opensource.org/licenses): https://opensource.org/license/apache-2-0
- [x] License correctly listed in package.xmls
- [x] Public source repo: https://github.com/barry-Xu-2018/ros2_load_balancing_service/
- [x] Source repository contains ROS packages
- [ ] Each package meets [REP-144](https://www.ros.org/reps/rep-0144.html) naming conventions

No strong objections to the name but just floating some ideas for discussion. Is the suffix `_service` indicating that the executable provides the "service" of load balancing or that the executable balances load on ROS 2 services? Some suggestions to disambiguate
- `ros2_service_load_balancer`
- `service_load_balancer`

Would also like to hear from @sloretz before merging.

**Comment by clalancette** [comment]
> * 
> 
> 
> No strong objections to the name but just floating some ideas for discussion. Is the suffix `_service` indicating that the executable provides the "service" of load balancing or that the executable balances load on ROS 2 services? Some suggestions to disambiguate

Yeah, I think this is a good suggestion.  It's kind of hard to tell what the overloaded term "service" means in this context.

**Comment by Barry-Xu-2018** [comment]
Okay. I will use service_load_balancer instead of load_balancing_service.

**Comment by Barry-Xu-2018** [comment]
The failure is 
```
    def test_verify_files_identical():
        with Fold() as fold:
            print("""Checking if index.yaml and all referenced files comply to the formatting rules.
    If this fails you can run 'rosdistro_reformat index.yaml' to help cleanup.
    'rosdistro_reformat' shows the diff between the current files and their expected formatting.
    """)
```
But I don't know to execute ````rosdistro_reformat index.yaml```` in check actions.   
@Yadunund I changed package name and ran into this problem.  Could you help me ?

**Review by clalancette — APPROVED** [review]
I'm OK with this now, but would like to get a confirmation from @Yadunund before merging.

**Review by fujitatomoya — APPROVED** [review]

**Comment by fujitatomoya** [comment]
@Yadunund friendly ping.

**Review by Yadunund — APPROVED** [review]

**Comment by clalancette** [comment]
> After checking error, I think ros-rolling-rclcpp: 29.0.0-1noble.20241015.213657 used in build environment doesn't include this PR [ros2/rclcpp#2614](https://github.com/ros2/rclcpp/pull/2614) (merged in 10/04).

Yesterday I did a round of releases, and that included a new rclcpp release.  So it should rebuild (in point of fact, it should have already rebuilt) against rclcpp-29.1.0, unless something went wrong.  I'll take a look later.

**Comment by fujitatomoya** [comment]
@clalancette thanks for helping us on this.

**Pull request merged**